types/aws-lambda/trigger/cdk-custom-resource.d.ts

```diff
@@ -0,0 +1,65 @@
+import { CloudFormationCustomResourceEvent } from './cloudformation-custom-resource';
+import { Handler } from '../handler';
+
+// The CDK docs only specify 'important' properties, but in reality the incoming event
+// to the Lambda matches that of a traditional custom resource.
+// This includes the ResponseURL property which should not be used as the framework
+// itself will deal with delivering responses.
+export type CdkCustomResourceEvent = CloudFormationCustomResourceEvent & {
+    /**
+     * **This URL should not be used.** The CDK Provider Framework will call this URL
+     * automatically based on the response produced by the Lambda handler.
+     */
+    ResponseURL: string;
+};
+
+/**
+ * A custom resource based on the AWS CDK custom resource Provider Framework.
+ * This is not to be confused with traditional CloudFormation custom resources.
+ * @link https://docs.aws.amazon.com/cdk/api/latest/docs/custom-resources-readme.html#handling-lifecycle-events-onevent
+ */
+export type CdkCustomResourceHandler = Handler<CdkCustomResourceEvent, CdkCustomResourceResponse>;
```

Comment:
It's a bit obsolete, but you're missing the `export type CdkCustomResourceCallback = Callback<CdfCustomResourceResponse>;` that triggers with a response define - I notice we don't actually document this in index.d.ts, which is an issue, but I doubt they get much usage nowadays. (Probably time for a major version break where we can strip out all the compatibility definitions and make things a bit cleaner)